water_neuron/src/management.rs

```diff
@@ -249,6 +249,21 @@ pub async fn get_neuron_ids() -> Result<Vec<u64>, String> {
     }
 }
 
+pub async fn get_neuron_info(id: u64) -> Result<Result<NeuronInfo, GovernanceError>, String> {
+    let result: Result<(Result<NeuronInfo, GovernanceError>,), (i32, String)> =
+        ic_cdk::api::call::call(NNS_GOVERNANCE_ID, "get_neuron_info", (id,))
+            .await
+            .map_err(|(code, msg)| (code as i32, msg));
+
+    match result {
+        Ok((res,)) => Ok(res),
+        Err((code, msg)) => Err(format!(
+            "Error while calling Governance canister ({}): {:?}",
+            code, msg
+        )),
+    }
+}
+
```

Comment:
?

Reply:
I made that because the MergeResponse gives only the current stakes of the target neuron and the source neuron (which is 0 if the call succeeded) however we have to make a transfer of nICP to the user depending on the old stake of the source neuron so this endpoint was made to fetch it.